boards/components/src/adc.rs

```diff
@@ -27,7 +27,6 @@ macro_rules! adc_component_static {
 macro_rules! adc_syscall_component_helper {
     ($($P:expr),+ $(,)?) => {{
         use kernel::count_expressions;
-        use kernel::static_init;
```

Comment:
Please keep this line, as the macro might be used in places where `static_init` is not defined.